Vungle/VungleAdapter/ALVungleMediationAdapter.m

```diff
@@ -207,19 +207,27 @@ - (void)loadInterstitialAdForParameters:(id<MAAdapterResponseParameters>)paramet
     
     if ( ![VungleAds isInitialized] )
     {
-        [self log: @"Vungle SDK not successfully initialized: failing interstitial ad load..."];
-        [delegate didFailToLoadInterstitialAdWithError: MAAdapterError.notInitialized];
+        [self initializeWithParameters:(id<MAAdapterInitializationParameters>)parameters completionHandler:^(MAAdapterInitializationStatus initializationStatus, NSString * _Nullable errorMessage) {
+            if ( initializationStatus == MAAdapterInitializationStatusInitializedFailure ) {
+                [self log: @"Vungle SDK not successfully initialized: failing interstitial ad load..."];
+                [delegate didFailToLoadInterstitialAdWithError: MAAdapterError.notInitialized];
+            } else if ( initializationStatus == MAAdapterInitializationStatusInitializedSuccess ) {
+                self.interstitialAd = [[VungleInterstitial alloc] initWithPlacementId: placementIdentifier];
```

Comment:
I think you should separate this initialize part into a different method so you can avoid duplicating L215-L219 into L225-L229

Reply:
updated